docs/content/technical/devkitarm_porting_guide.md

```diff
@@ -222,3 +212,51 @@ In BlocksDS, `scanKeys()` is used to latch the current state of the keys and the
 touch screen. This forces the developer to call `scanKeys()`, but it also
 ensures that there are no race conditions, as `scanKeys()` will read all the
 state atomically.
+
+## 6. Updating legacy devkitARM homebrew
+
+Occasionally, various components of the devkitARM toolchain introduced breaking
+changes of their own, which lead to the unfortunate situation of homebrew being
+stuck on legacy, buggier and less reliable versions of toolchains. These issues
+also need to be resolved before updating to BlocksDS. Some known issues and their
+remedies are documented here; note that this list is not exhaustive.
+
+### New assembly function definition syntax
+
+For ARM/Thumb interwork (calling ARM functions from Thumb and vice versa) to work
+correctly in more recent versions of the GNU toolchain, functions must be
+appropriately marked in the generated object file. While they are annotated 
+automatically for code generated by GCC, it has to be done manually for assembly
+code. For example, a snippet of assembly as follows:
+
+```asm
+   .global myFunction
+myFunction:
+```
+
+should be updated to read as follows:
+
+```asm
+   .global myFunction
+   .type myFunction, %function
+myFunction:
+```
```

Comment:
Maybe mention the macros of this file: https://github.com/blocksds/libnds/blob/d793d51d4a960d282a9fd1f95921472e567a1f8b/include/nds/asminc.h